### PR TITLE
Nanopore fast5 upload and display interface

### PR DESCRIPTION
**Pull request by joshsadam**
## Description of changes
Updated the Sample > Files interface to allow for the upload and display of FAST5 files.

## Related issue
N/A

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~~
~~* [ ] Tests added (or description of how to test) for any new features.~~
~~* [ ] User documentation updated for UI or technical changes.~~

### CODE REVIEW

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/SamplePairer.java [hunk 1]
```diff
@@ -12,136 +13,136 @@
...
+                                    && Stream.of(reverseMatches).anyMatch(x -> file2ID.contains(x)))
+                                    || (Stream.of(reverseMatches).anyMatch(x -> file1ID.contains(x))
+                                    && Stream.of(forwardMatches).anyMatch(x -> file2ID.contains(x)))) {
+                                filePair = new MultipartFile[]{file1, file2};
+                            }
+                        }
+
+                        if (filePair != null) {
+                            pair = true;
+                            //organizedFiles.put(diffs.get(0).text, Arrays.asList(filePair));
```
Comment (tom114) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/SamplePairer.java:96]: Can you pull this commented code.  Sorry that's on me I had them in my other branch.

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/SamplePairer.java [hunk 2]
```diff
@@ -12,136 +13,136 @@
...
+
+                            logger.trace("Uploaded files [" + filePair[0].getName() + ", " + filePair[1].getName()
+                                    + "] were paired.");
+                            wasChecked.add(file2);
+                        }
+                    }
+                }
+                if (!pair) {
+                    MultipartFile[] singleFile = {file1};
+                    //organizedFiles.put(file1.getOriginalFilename(), Arrays.asList(singleFile));
```
Comment (tom114) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/SamplePairer.java:108]: Same here.  Delete it please.

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/SamplesController.java [hunk 3]
```diff
@@ -286,16 +287,24 @@ public String getSampleFiles(final Model model, @PathVariable Long projectId, @P
...
-		Collection<SampleSequencingObjectJoin> singleFileJoins = sequencingObjectService
-				.getSequencesForSampleOfType(sample, SingleEndSequenceFile.class);
+		Collection<SampleSequencingObjectJoin> filePairJoins = sequencingObjectService.getSequencesForSampleOfType(
+				sample, SequenceFilePair.class);
+		Collection<SampleSequencingObjectJoin> singleFileJoins = sequencingObjectService.getSequencesForSampleOfType(
+				sample, SingleEndSequenceFile.class);
+		Collection<SampleSequencingObjectJoin> fast5FileJoins = sequencingObjectService.getSequencesForSampleOfType(
+				sample, Fast5Object.class);
+		List<Fast5Object> fast5 = fast5FileJoins.stream()
+				.map(j -> (Fast5Object) j.getObject())
```
Comment (tom114) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/SamplesController.java:297]: No need to cast here.  You're not doing anything with them after they're cast. It cna just be a List of `SequencingObject` I think.

File: src/main/webapp/pages/samples/sample_files.html [hunk 4]
```diff
@@ -182,6 +182,58 @@ <h4 class="media-heading" th:text="#{samples.files.empty.title}"></h4>
 			</table>
 		</div>
 	</div>
+
+	<div class="row" th:if="${not #lists.isEmpty(fast5)}">
```
Comment (tom114) [inline src/main/webapp/pages/samples/sample_files.html:186]: If we have fast5 files in here it still shows the "No sequencing data" message.  This should be hidden if there are fast5 files.

image

File: src/main/webapp/pages/samples/sample_files.html [hunk 5]
```diff
@@ -182,6 +182,58 @@ <h4 class="media-heading" th:text="#{samples.files.empty.title}"></h4>
...
+			<h2 th:text="#{sample.files.fast5}">FAST5 Files</h2>
+			<table class="table fast5" th:with="dateFormat=#{locale.date.short}, url=@{${#httpServletRequest.requestURI}}">
+				<tr class="fast5_row" th:each="file : ${fast5}">
+					<td class="td_icon fast5_icon">
+						<i class="far fa-file-alt"></i>
+					</td>
+					<td>
+						<div class="file-name td_info--value"
+							 style="display: flex; align-items: center; height: 35px;">
+							<div th:text="${file.label}"></div>
```
Comment (tom114) [inline src/main/webapp/pages/samples/sample_files.html:197]: Need the link to the fastqc results here.  It's exaclty the same as for sequence files.

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/SamplePairer.java (hunk 1 above)
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/SamplePairer.java:96]: Updated in ea379e563c249e785d87b7b7906dd0ef1aa37909

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/SamplePairer.java (hunk 2 above)
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/SamplePairer.java:108]: Updated in ea379e563c249e785d87b7b7906dd0ef1aa37909

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/SamplesController.java (hunk 3 above)
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/samples/SamplesController.java:297]: Updated in 0d74082a0fff036d27cdfcfa1007c91abbd9cee6

File: src/main/webapp/pages/samples/sample_files.html (hunk 4 above)
Comment (joshsadam) [reply src/main/webapp/pages/samples/sample_files.html:186]: Updated in 04448d4847cb221c0c8cc7f3cffd2e242c5054ed

File: src/main/webapp/pages/samples/sample_files.html (hunk 5 above)
Comment (joshsadam) [reply src/main/webapp/pages/samples/sample_files.html:197]: Updated in 82d3f5f5397234fcc0b347bf1e8f669743627243

File: src/main/webapp/pages/samples/sample_files.html [hunk 6]
```diff
@@ -182,6 +182,58 @@ <h4 class="media-heading" th:text="#{samples.files.empty.title}"></h4>
 			</table>
 		</div>
 	</div>
+
+	<div class="row" th:if="${not #lists.isEmpty(fast5)}">
+		<div class="col-md-12">
+			<h2 th:text="#{sample.files.fast5}">FAST5 Files</h2>
+			<table class="table fast5" th:with="dateFormat=#{locale.date.short}, url=@{${#httpServletRequest.requestURI}}">
```
Comment (tom114) [inline src/main/webapp/pages/samples/sample_files.html:189]: do we need this `url` variable for anything?  doesn't seem to be used anywhere

Comment (joshsadam) [reply src/main/webapp/pages/samples/sample_files.html]: Updated in fbbf5f28d102ec8438c3e2ce9477e8ff2a9be5cf

**Pull request merged**